deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -384,6 +377,18 @@ def __init__(self,
             data_parallel_partitions = self.get_data_parallel_partitions(self.bit16_groups_flat[i], i)
             self.parallel_partitioned_bit16_groups.append(data_parallel_partitions)
 
+            # Record padding required for alignment
+            left_boundary = sum([t.numel() for t in data_parallel_partitions[:partition_id]])
+            curr_partition_size = data_parallel_partitions[partition_id].numel()
+
+            if orig_group_numel <= left_boundary:
+                padding = curr_partition_size
+            elif orig_group_numel < left_boundary + curr_partition_size:
+                padding = orig_group_numel - left_boundary
```

Comment:
Please double check that it shouldn't be 
```suggestion
                padding = (left_boundary + curr_partition_size) - orig_group_numel
```

Reply:
Yes, you are right, the padding should be the right side of this partition.